src/lib/promptToEdit.tsx

```diff
@@ -37,43 +54,117 @@ function convertAppRangeToApiRange(
   }
 }
 
+type TextToCadErrorResponse = {
+  error_code: string
+  message: string
+}
+
+async function submitTextToCadRequest(
+  body: {
+    prompt: string
+    source_ranges: Models['SourceRangePrompt_type'][]
+    project_name?: string
+    kcl_version: string
+  },
+  files: KittyCadLibFile[],
+  token: string
+): Promise<TextToCadMultiFileIteration_type | Error> {
+  const formData = new FormData()
+  formData.append('body', JSON.stringify(body))
+
+  files.forEach((file) => {
+    formData.append('files', file.data, file.name)
+  })
+
+  const response = await fetch(
+    `${VITE_KC_API_BASE_URL}/ml/text-to-cad/multi-file/iteration`,
+    {
+      method: 'POST',
+      headers: {
+        Authorization: `Bearer ${token}`,
+      },
+      body: formData,
+    }
+  )
+
+  if (!response.ok) {
+    return new Error(`HTTP error! status: ${response.status}`)
+  }
+
+  const data = await response.json()
+  if ('error_code' in data) {
+    const errorData = data as TextToCadErrorResponse
+    return new Error(errorData.message || 'Unknown error')
+  }
+
+  return data as TextToCadMultiFileIteration_type
```

Comment:
Could probably use `satisfies` instead of `as` here to avoid coercion

Reply:
@franknoirot do you still want this fixed up?